src/lgcompressed.jl

```diff
@@ -0,0 +1,41 @@
+struct LGCompressedFormat <: AbstractGraphFormat end
+
+function savegraph(fn::AbstractString, g::AbstractGraph, gname::AbstractString,
+    format::LGCompressedFormat)
+    io = open(fn, "w")
+    try
+        io = GzipCompressorStream(io)
+        return savegraph(io, g, gname, LightGraphs.LGFormat())
+    catch
+        rethrow()
+    finally
+        close(io)
+    end
+end
+
+savegraph(fn::AbstractString, g::AbstractGraph, format::LGCompressedFormat) =
+    savegraph(fn, g, "graph", format)
+
+savegraph(fn::AbstractString, g::AbstractGraph; compress) = savegraph(fn, g, LGCompressedFormat())
```

Comment:
Why do we need the keyword argument here?

Reply:
backwards compatibility with the old version of persistence, which had a default keyword. This allows dispatching.